robolectric/src/test/java/org/robolectric/shadows/ShadowBatteryManagerTest.java

```diff
@@ -0,0 +1,74 @@
+package org.robolectric.shadows;
+
+import android.content.Context;
+import android.os.BatteryManager;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.TestRunners;
+import org.robolectric.annotation.Config;
+
+import static android.os.Build.VERSION_CODES.LOLLIPOP;
+import static android.os.Build.VERSION_CODES.M;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(TestRunners.MultiApiSelfTest.class)
+public class ShadowBatteryManagerTest {
+  private BatteryManager batteryManager;
+  private ShadowBatteryManager shadowBatteryManager;
+  private static final int TEST_ID = 123;
+
+  @Before
+  public void before() {
+    batteryManager = (BatteryManager) RuntimeEnvironment.application.getSystemService(
+        Context.BATTERY_SERVICE);
+    shadowBatteryManager = shadowOf(batteryManager);
+  }
+
+  @Test
+  @Config(minSdk = M)
+  public void testIsCharging() {
+    assertThat(batteryManager.isCharging()).isFalse();
+
+    shadowBatteryManager.setIsCharging(true);
+
+    assertThat(batteryManager.isCharging()).isTrue();
+
+    shadowBatteryManager.setIsCharging(false);
+
+    assertThat(batteryManager.isCharging()).isFalse();
+  }
+
+  @Test
+  @Config(minSdk = LOLLIPOP)
```

Comment:
You could always put  @Config(minSdk = LOLLIPOP) on the class which might be better as it shows more clearly that BatteryManager was introduced in L  (method annotations override class, for the M features added)